examples/pytorch/ogb/ogbn-mag/hetero_rgcn.py

```diff
@@ -383,27 +391,49 @@ def main(args):
 
     for run in range(args.runs):
 
-        embed_layer.reset_parameters()
-        model.reset_parameters()
+        try:
+            embed_layer.reset_parameters()
+            model.reset_parameters()
+        except:
+            # old pytorch version doesn't support reset_parameters() API
+            pass
 
         # optimizer
         all_params = itertools.chain(
             model.parameters(), embed_layer.parameters()
         )
         optimizer = th.optim.Adam(all_params, lr=0.01)
 
-        logger = train(
-            g,
-            model,
-            embed_layer,
-            optimizer,
-            train_loader,
-            split_idx,
-            labels,
-            logger,
-            device,
-            run,
-        )
+        if args.num_workers != 0 and device == "cpu" and is_support_affinity(v_t):
```

Comment:
@chang-l , only when `num_workers `are configured, will do `enable_cpu_affinity `here. Which prevent previous method of using full system cores, and I tried using 24 here, performance is quite similar to using 48, which shows around 20% benefit comparing with not using `enable_cpu_affinity `

Reply:
Thanks. I still don't understand why you need loader_cores and compute_cores specified in `enable_cpu_affinity()` at L411. Can you remove L408-L411 and replace with `with train_loader.enable_cpu_affinity()`?
Also, can you pls verify that it is still working when GPU is available (device = 'cuda')?

Reply:
@chang-l , when replacing with `with train_loader.enable_cpu_affinity()` and set num_workers = 24, it will go crash as the screenshot I pasted last time. That is why `loader_cores` and `compute_cores` are specified.
![image](https://user-images.githubusercontent.com/4355494/195368358-b0871237-a35d-4181-9e1d-3b1ded8afdad.png)

And yes, GPU is working, the PR is both for GPU and CPU performance ceoptimization, for GPU performance improved by 3x withL322 and L382. I added a switch at L407, so GPU won't go `enable_cpu_affinity()`